var/spack/repos/builtin/packages/acfl/package.py

```diff
@@ -276,3 +278,12 @@ def setup_run_environment(self, env):
         env.prepend_path("PATH", join_path(arm_dir, "bin"))
         env.prepend_path("CPATH", join_path(arm_dir, "include"))
         env.prepend_path("MANPATH", join_path(arm_dir, "share", "man"))
+
+    @run_after("install")
+    def check_install(self):
```

Comment:
If you want this to only run when passing `--test` option (see https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-installation-phase-test)), you'll want to add the package attributes directive before the method definition.  There is an example after the table at the link above.

```suggestion
    @on_package_attributes(run_tests=True)
    def check_install(self):
```

Reply:
I had tried adding `@on_package_attributes(run_tests=True)`, but it does not work when passing `--test` option. So, for now I would like this to always run post-installation.

Reply:
Did you pass it as `--test=root`?

Reply:
Yes. I observed the same behavior with some other packages as well. Perhaps I can try to reproduce it again and raise an issue.

Reply:
Up to you.  If you want the tests to run after every install and, ideally, they run quickly, then we could leave this as-is.

You could also consider switching the checks to be stand-alone, post-install checks as described https://spack.readthedocs.io/en/latest/packaging_guide.html#stand-alone-or-smoke-tests.  Then they would only be performed -- against the installed software -- when someone runs e.g. `spack test run acfl`.

Reply:
These tests run rather quickly. So, I would like to leave it as-is. I will find time to try to reproduce the problem I saw with `@on_package_attributes(run_tests=True)` and raise an issue, if I can reproduce. 

I had considered making the stand-alone check as well (hence the leftover line above), but that is perhaps for another commit.